### PR TITLE
Provide DSL for echoing text/bytes into a process

### PR DESCRIPTION
**Pull request by jroper**
Implements https://github.com/scala/bug/issues/11668

### CODE REVIEW

File: src/library/scala/sys/process/Process.scala [hunk 1]
```diff
@@ -172,6 +174,50 @@ trait ProcessCreation {
...
+    * echo("Hello, world!") #> "curl -d - https://example.com/upload" !
+    * }}}
+    *
+    * Unlike the shell echo command, this does not append a newline to the text.
+    *
+    * @param text The text to echo.
+    */
+  def echo(text: String): Source = {
+    new ProcessBuilder.IStreamBuilder(
+      new ByteArrayInputStream(text.getBytes()),
```
Comment (martijnhoekstra) [inline src/library/scala/sys/process/Process.scala:196]: Is using the platform encoding correct here, or should it take an encoding as an argument?

File: src/library/scala/sys/process/Process.scala [hunk 2]
```diff
@@ -172,6 +174,50 @@ trait ProcessCreation {
...
+  /**
+    * Echo the given text.
+    *
+    * This can be used as the source of input for a process, for example:
+    *
+    * {{{
+    * import scala.sys.process._
+    * import scala.sys.process.Process.echo
+    *
+    * echo("Hello, world!") #> "curl -d - https://example.com/upload" !
```
Comment (julienrf) [inline src/library/scala/sys/process/Process.scala:187]: Postfix syntax now requires importing `scala.language.postfixOps`. I suggest the following:

```suggestion
    * echo("Hello, world!") #> "curl -d - https://example.com/upload".!
```

File: src/library/scala/sys/process/Process.scala [hunk 3]
```diff
@@ -172,6 +174,50 @@ trait ProcessCreation {
...
+  /**
+    * Echo the given text.
+    *
+    * This can be used as the source of input for a process, for example:
+    *
+    * {{{
+    * import scala.sys.process._
+    * import scala.sys.process.Process.echo
+    *
+    * echo(someBytes) #> "curl -d - https://example.com/upload" !
```
Comment (julienrf) [inline src/library/scala/sys/process/Process.scala:210]:
```suggestion
    * echo(someBytes) #> "curl -d - https://example.com/upload".!
```

File: src/library/scala/sys/process/Process.scala (hunk 1 above)
Comment (jroper) [reply src/library/scala/sys/process/Process.scala:196]: I think platform encoding is correct. I did consider taking an implicit for the encoding, but the process API already assumes platform encoding everywhere else that it handles text, eg all the output handling (eg, `.!!`) uses the platform encoding, and in shell scripting in general, platform encoding is assumed, eg by all the unix commands that handle text (grep, sed, etc).

File: src/library/scala/sys/process/Process.scala (hunk 2 above)
Comment (som-snytt) [reply src/library/scala/sys/process/Process.scala:187]: You'd need parens. I think one must suck it up, so to speak. I don't know if they intend to eliminate postfix, if the language import is just the methadone of syntax and they're actually going to cut us off, but this has to be the poster child for keeping us hooked.

**Comment by som-snytt** [comment]
New API would have to wait for the process module spin-off, which might be its last "Harrah", as it were.

**Comment by dwijnand** [comment]
Shall we park (read: close) this, for now, using https://github.com/scala/bug/issues/11668 as the beacon for possibly recovering this work one day?

**Comment by lrytz** [comment]
Agree.

**Pull request closed without merging**